demos/door_manipulation_demo/src/door_open_as.cpp

```diff
@@ -0,0 +1,524 @@
+#include <signal.h>
+#include <vector>
+#include <string>
+#include <sys/stat.h>
+#include <ros/ros.h>
+#include <ros/package.h>
+#include <vector>
+#include <math.h>
+#include <cstdlib>
+#include <std_msgs/String.h>
+
+#include <sensor_msgs/PointCloud2.h>
+
+#include <Eigen/Dense>
+#include <eigen_conversions/eigen_msg.h>
+
+#include <sensor_msgs/JointState.h>
+#include <geometry_msgs/TwistStamped.h>
+#include <geometry_msgs/PoseArray.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <std_msgs/Float32.h>
+
+#include <visualization_msgs/Marker.h>
+#include <visualization_msgs/MarkerArray.h>
+
+
+// PCL specific includes
+#include <pcl/conversions.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/point_cloud.h>
+#include <pcl/console/parse.h>
+#include <pcl/point_types.h>
+#include <pcl/io/openni_grabber.h>
+#include <pcl/sample_consensus/sac_model_plane.h>
+#include <pcl/common/time.h>
+#include <pcl/common/common.h>
+
+//actions
+#include <actionlib/client/simple_action_client.h>
+#include "jaco_msgs/SetFingersPositionAction.h"
+#include "jaco_msgs/ArmPoseAction.h"
+#include "jaco_msgs/ArmJointAnglesAction.h"
+
+#include <pcl/kdtree/kdtree.h>
+#include <pcl_ros/impl/transforms.hpp>
+
+//including package services 
+#include "door_manipulation_demo/door_perception.h"
+
+
+#include <tf/transform_listener.h>
+#include <tf/tf.h>
+
+#include <moveit_msgs/DisplayRobotState.h>
+// Kinematics
+#include <moveit_msgs/GetPositionFK.h>
+#include <moveit_msgs/GetPositionIK.h>
+
+#include <moveit_utils/AngularVelCtrl.h>
+#include <moveit_utils/MicoMoveitJointPose.h>
+#include <moveit_utils/MicoMoveitCartesianPose.h>
+#include <geometry_msgs/TwistStamped.h>
+#include <std_msgs/Header.h>
+#include <time.h>
+#include <stdint.h>
+#include <segbot_arm_manipulation/arm_utils.h>
+#include <segbot_arm_manipulation/grasp_utils.h>
+#include <agile_grasp/Grasps.h>
+#include <door_manipulation_demo/PushDoorAction.h>
+#include <actionlib/server/simple_action_server.h>
+#include <geometry_msgs/TwistStamped.h>
+
+#define NUM_JOINTS 8
```

Comment:
The arm has 8 joints so i use it to check if that current joint state of the arm is. If this reads false I can see their is something going wrong in getting the arm's details.